odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -1603,6 +1603,26 @@ def test_60_x2many_domain(self):
         self.assertFalse(discussion.very_important_messages)
         self.assertFalse(message.exists())
 
+    def test_60_many2many_domain(self):
+        """ test the cache consistency of a many2many field with a domain """
+        discussion = self.env.ref('test_new_api.discussion_0')
+        category = self.env['test_new_api.category'].create({'name': "Foo"})
+        discussion.categories = category
+        discussion.flush()
+        discussion.invalidate_cache()
+
+        # patch the many2many field to give it a domain
+        field = discussion._fields['categories']
+        self.patch(field, 'domain', [('color', '!=', 42)])
+        self.registry.setup_models(self.cr)
```

Comment:
why is the domain patched on-the-fly? is it part of the test case (like adding domain through inheritance) or just to avoid recreating a separate set of test models that already have an m2m field with a domain?

Reply:
This is just to avoid recreating test models.  I picked one with a many2many field, and added a domain on it.